cdci_data_analysis/analysis/email_helper.py

```diff
@@ -0,0 +1,220 @@
+    # TODO: should get from pkgresources or so
+    env = Environment(loader=FileSystemLoader('%s/../flask_app/templates/' % os.path.dirname(__file__)))
+    env.filters['timestamp2isot'] = timestamp2isot
+
+    api_code_no_token = re.sub('"token": ".*?"', '"token": "<PLEASE-INSERT-YOUR-TOKEN-HERE>"', api_code)\
+                        .strip()\
+                        .replace("\n", "<br>\n")
+    
+    email_data = {
+        'oda_site': { 
+            #TODO: get from config
+            'site_name': 'University of Geneva',
+            'frontend_url': 'https://www.astro.unige.ch/mmoda', 
+            'contact': 'contact@odahub.io'
+        },
+        'request': {
+            'job_id': job_id,
+            'status': status,
+            'instrument': instrument,
+            'product_type': product_type,
+            'time_request': time_request,
+            'request_url': request_url,
+            'api_code_no_token': api_code_no_token,
+        }
+    }
+
+    template = env.get_template('email.html')
+    email_body_html = template.render(**email_data)
+    
+    email_subject = re.search("<title>(.*?)</title>", email_body_html).groups()[0]    
+
+    email_text = textify_email(email_body_html)
+    
+    server = None
+    logger.info("Sending email")
+    # Create the plain-text and HTML version of your message,
+    # since emails with HTML content might be, sometimes, not supported
+        
+    try:
+        # send the mail with the status update to the mail provided with the token
+        # eg done/failed/submitted
+        # test with the local server
+        smtp_server = config.smtp_server
+        port = config.smtp_port
+        sender_email_address = config.sender_email_address
+        cc_receivers_email_addresses = config.cc_receivers_email_addresses
+        receiver_email_address = tokenHelper.get_token_user_email_address(decoded_token)
+        receivers_email_addresses = [receiver_email_address] + cc_receivers_email_addresses
+        # creation of the message
+        message = MIMEMultipart("alternative")
+        message["Subject"] = email_subject
+        message["From"] = sender_email_address
+        message["To"] = receiver_email_address
+        message["CC"] = ", ".join(cc_receivers_email_addresses)
+
+        part1 = MIMEText(email_text, "plain")
+        part2 = MIMEText(email_body_html, "html")
+        message.attach(part1)
+        message.attach(part2)
+
+        smtp_server_password = config.smtp_server_password
+        # Create a secure SSL context
+        context = ssl.create_default_context()
+        #
+        # Try to log in to server and send email
+        server = smtplib.SMTP(smtp_server, port)
+        # just for testing purposes, not ssl is established
+        if smtp_server != "localhost":
+            try:
+                server.starttls(context=context)
+            except Exception as e:
+                logger.warning(f'unable to start TLS: {e}')
```

Comment:
Also, need an option for when the server requires SSL over 465 from the beginning, by using SMTP_SSL 
In this case, simple SMTP may just be stuck for a long time.

Reply:
do you have a place where this is run? if yes and you need it, please make an issue.